src/persistent-configuration-cache.ts

```diff
@@ -0,0 +1,67 @@
+import { logger } from './application-logger';
+import { Configuration } from './configuration';
+import { ConfigurationFeed, ConfigurationSource } from './configuration-feed';
+
+/**
+ * Persistent configuration storages are responsible for persisting
+ * configuration between SDK reloads.
+ */
+export interface PersistentConfigurationStorage {
+  /**
+   * Load configuration from the persistent storage.
+   *
+   * The method may fail to load a configuration or throw an
+   * exception (which is generally ignored).
+   */
+  loadConfiguration(): PromiseLike<Configuration | null>;
+
+  /**
+   * Store configuration to the persistent storage.
+   *
+   * The method is allowed to do async work (which is not awaited) or
+   * throw exceptions (which are ignored).
+   */
+  storeConfiguration(configuration: Configuration | null): PromiseLike<void>;
```

Comment:
I haven't finished reading all the code yet but something to call out here if it's not already integrated, is the need for the persistent store to be keyed by API key (or some other function thereof) so that a single storage subsystem can store config for multiple api keys simultaneously.

Reply:
I think that's best handled by individual constructors. We don't allow changing sdk key on the fly, so a persistent store instance is always used with the same key

Reply:
Does the upstream creation have a way to not have these configurations clobber each other if different keys are used?